src/jsonLogic.js

```diff
@@ -0,0 +1,108 @@
+import jsonLogic from 'json-logic-js';
+
+/**
+ * Parses the JSON schema to extract the advanced validation logic and returns a set of functionality to check the current status of said rules.
+ * @param {Object} schema - JSON schema node
+ * @param {Object} initialValues - form state
+ * @returns {Object}
+ */
+export function createValidationChecker(schema) {
+  const scopes = new Map();
+
+  function createScopes(jsonSchema, key = 'root') {
+    scopes.set(key, createValidationsScope(jsonSchema));
+    Object.entries(jsonSchema?.properties ?? {})
+      .filter(([, property]) => property.type === 'object' || property.type === 'array')
+      .forEach(([key, property]) => {
+        if (property.type === 'array') {
+          createScopes(property.items, `${key}[]`);
+        }
+        createScopes(property, key);
+      });
```

Comment:
Do we need two scopes for arrays?  In this case, we'll have the `array-x[]` and `array-x` scope, right?

Reply:
you're right, good catch